tests/python/test_exception.py

```diff
@@ -84,3 +84,61 @@ def foo():
 TypeError: 'NoneType' object is not callable"""
     print(e.value.args[0])
     assert e.value.args[0] == msg
+
+
+@ti.test()
+def test_tab():
+    frameinfo = getframeinfo(currentframe())
+    with pytest.raises(ti.TaichiCompilationError) as e:
+        # yapf: disable
+        @ti.kernel
+        def foo():
+            a(11,	22,	3)
+        foo()
+        # yapf: enable
+    lineno = frameinfo.lineno
+    file = frameinfo.filename
+    if version_info < (3, 8):
+        msg = f"""\
+On line {lineno + 5} of file "{file}":
+            a(11,   22, 3)
```

Comment:
I'm a little curious how do we determine the number of spaces here. IMHO the number of spaces shown depends on editor configurations. For example, in my browser this line doesn't align with the function definition. What is the goal we aim to achieve?

Reply:
We have to convert tabs to spaces in the source code when printing because otherwise we cannot generate the hint line accurately. As python allows to indent using tab instead of 4 spaces, I just assumed that the tabsize of python program is 4. Or should we make this configurable?

Reply:
What is the advantage of using `textwrap.fill` over replacing `\t` with 4 spaces?

Reply:
Updated the example in the issue. Also, this is the right way to treat tabs, and replacing `\t` with 4 spaces is simply wrong.